### PR TITLE
Add test for GrafanaDashboard presence

### PR DESCRIPTION
**Pull request by mdujava**
This is new PR, renaming the branch closed original PR.

### CODE REVIEW

File: testsuite/tests/grafana/test_dashboards.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        pytest.mark.skipif("TESTED_VERSION < Version('2.9')"),
+        pytest.mark.required_capabilities(Capability.OCP4),
+    ]
+
+# pylint: disable=line-too-long
+DASHBOARDS = [
+    ("apicast-mainapp",
+     "sum(rate(nginx_http_connections{namespace='$namespace', pod=~'apicast-$env.*'}[1m])) by (state)"),
+    ("apicast-services",
+     "sum(rate(upstream_status{namespace='$namespace', pod=~'apicast-$env.*', service_id='$service_id'}[1m])) by (status)"),  # noqa: E501
```
Comment (pehala) [inline testsuite/tests/grafana/test_dashboards.py:16]: Not sure if it won't degrade the readability, but if you want to remove the long lines you can use this method of splitting it https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation.

File: testsuite/tests/grafana/test_dashboards.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    ("backend",
+     "sum(rate(apisonator_listener_response_codes{namespace=\\\"$namespace\\\",request_type=\\\"authorize\\\"}[1m]))"),  # noqa: E501
```
Comment (pehala) [inline testsuite/tests/grafana/test_dashboards.py:18]: If you use `'` you can get rid of at least one \ for each escaped quote.

File: testsuite/tests/grafana/test_dashboards.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+     "sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate{namespace=\\\"$namespace\\\", pod=\\\"$pod\\\", container!=\\\"POD\\\", cluster=\\\"$cluster\\\"}) by (container)"),  # noqa: E501
+    ("system",
+     "sum(rate(rails_requests_total{namespace='$namespace',pod=~'system-app-[a-z0-9]+-[a-z0-9]+'}[1m])) by (status)"),
+    ("zync",
+     "sum(rate(rails_requests_total{namespace='$namespace',pod=~'zync-[a-z0-9]+-[a-z0-9]+'}[1m])) by (status)"),
+]
+# pylint: enable=line-too-long
+
+
+@pytest.mark.smoke
```
Comment (pehala) [inline testsuite/tests/grafana/test_dashboards.py:31]: Smokes need to be run even without any additional configuration of the instance, is it really the case for grafana?

File: testsuite/tests/grafana/conftest.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+"""Check for openshift configuration"""
+
+from weakget import weakget
+import pytest
+
+from testsuite.utils import warn_and_skip
+
+
+@pytest.fixture(scope="module", autouse=True)
+def require_openshift(testconfig):
```
Comment (pehala) [inline testsuite/tests/grafana/conftest.py:10]: Is this really the only thing that would skip this? Shouldnt there be a function checking if the grafana is even deployed and configured with the current instance?

File: testsuite/tests/grafana/test_dashboards.py (hunk 1 above)
Comment (mdujava) [reply testsuite/tests/grafana/test_dashboards.py]: I split the lines.

File: testsuite/tests/grafana/test_dashboards.py (hunk 2 above)
Comment (mdujava) [reply testsuite/tests/grafana/test_dashboards.py]: rewritten to use `'` in strings that contains `"`

File: testsuite/tests/grafana/test_dashboards.py (hunk 3 above)
Comment (mdujava) [reply testsuite/tests/grafana/test_dashboards.py:34]: this test is checking openshift object that are created by `monitoring > enabled : true` after openshift knows about grafana definitions.

I added check for the required configuration inside of `APIManager` object.

File: testsuite/tests/grafana/conftest.py (hunk 4 above)
Comment (mdujava) [reply testsuite/tests/grafana/conftest.py:10]: added one more requirement mentioned in prev comment.

**Review by pehala — APPROVED** [review]

**Review by mganisin — APPROVED** [review]

**Pull request merged**